src/backends/sqlite3/CMakeLists.txt

```diff
@@ -32,3 +34,31 @@ soci_define_backend_target(
 if (NOT SOCI_SQLITE3)
   return()
 endif()
+
+# If SQLite3 was found, this target must have been created.
+if (NOT TARGET SQLite::SQLite3)
+  # But if it wasn't, use our built-in version, after checking that it is
+  # available.
+  if (NOT EXISTS "${PROJECT_SOURCE_DIR}/3rdparty/sqlite3/sqlite3.c")
+    message(WARNING "SQLite3 not found and built-in version not available, have you cloned SOCI repository with --recurse-submodules?")
```

Comment:
Shouldn't this be a error rather than a warning? After all, if we reach this point we have enabled the backend which so far does not have the required link libraries configured. And if we don't have the submodule, this will guarantees compile and/or link errors, doesn't it?